### DISCUSSION TITLE
RESTDataSource GET with nested parameter

### DISCUSSION THREAD

**Discussion by lclankyo** [apollographql/apollo-server#6138] [Q&A]
Greetings,

I want to be able to do something like this:
`
const data = await this.get('person', {
	per_page: 1,
	filter: {
		name: "John",
		label: "Doe"
	},
});
`
It should produce a call
`/person?per_page=1&filter[name]=John&filter[label]=Doe`

I was struggling to find how to do this and was wondering if someone could point me in the right direction. Thanks!

**Comment by glasser** [comment]
Does `this.get('person', { per_page: 1, 'filter[name]': "John", 'filter[label]': "Doe"  })` solve your problem?

The params argument is just a standard web [`URLSearchParams` constructor argument](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams) so I don't think changing it to specially handle nested data in a non-standard way would make much sense. Are you working with somebody else's API or do you have control over the API yourself? I feel like it's a lot more common these days to just use a POSTed JSON body for complex nested parameters rather than trying to shoehorn them into the limited GET encoding...

**Reply by lclankyo** [reply]
Thanks a lot for the reply. Unfortunately, I do not have control to make it a post. 

I was looking at your suggestion, and that would require some more data transformation, which was something I was trying to avoid if possible. But if transforming the data some more is the only solution, then I will do that.
